File: Estruturas e bibliotecas/1110/1110.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		
		while(true) {
			int numeroDeCartas = scan.nextInt();
			Deque<Integer> pilha = new ArrayDeque<Integer>();
			String cartasDescartadas = "Discarded cards:";
			
			if(numeroDeCartas == 0) break;
			
			for(int numero=numeroDeCartas; numero>=1; numero--) {
				pilha.add(numero);
			}
			
			while(pilha.size() > 1) {
				cartasDescartadas += " " + pilha.removeLast() + ",";
				pilha.addFirst(pilha.removeLast());
			}
			
			if(numeroDeCartas >= 2) cartasDescartadas = cartasDescartadas.substring(0, cartasDescartadas.length()-1);
			
			System.out.println(cartasDescartadas);
			System.out.println("Remaining card: " + pilha.getFirst());
		}
	}
}

File: Estruturas e bibliotecas/1110/1110.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
	int tam, i;
	vector<int>::iterator it; 
    while(1){
    	
    	cin >> tam;
	    if(tam == 0) return 0;
	    
	    vector<int> pilha;
    	
	    for(i = 1; i<=tam; i++) pilha.push_back(i);
	    
	    it = pilha.begin();
	    
	    cout << "Discarded cards: ";
	    while(tam != 2){
	    	cout << *it << ", ";
	    	pilha.erase(it);
	    	it += 1;
	    	if(it > pilha.end()) it = pilha.begin()+1;
	    	else if(it == pilha.end()) it = pilha.begin();
	    	tam--;
		}
		cout << *it << endl;
		pilha.erase(it);
		cout << "Remaining card: " << pilha[0] << endl;
	}
}